tests/common/snappi_tests/common_helpers.py

```diff
@@ -105,9 +105,10 @@ def get_lossless_buffer_size(host_ans):
     Returns:
         total switch buffer size in byte (int)
     """
-    config_facts = host_ans.config_facts(host=host_ans.hostname,
+    dut_asic = host_ans.asic_instance()
```

Comment:
comment from @amitpawar12:    pls test T0 and T1 platform.

Reply:
Thanks for the comments. previous diff has error when using dut_asic.facts on T0/T1 devices. This is fixed now.
Following is the output on T1 Arista-7050CX3-32S-C32 platform.

ASIC: broadcom
ASIC Count: 1

```
(Pdb) list
109         config_facts = dut_asic.config_facts(host=host_ans.hostname,
110                                              source="running")['ansible_facts']
111         is_cisco8000_platform = True if 'cisco-8000' in host_ans.facts['platform_asic'] else False
112         breakpoint()
113  
114  ->     if "BUFFER_POOL" not in list(config_facts.keys()):
115             return None
116  
117         buffer_pools = config_facts['BUFFER_POOL']
118         profile_name = 'ingress_lossless_pool' if is_cisco8000_platform else 'egress_lossless_pool'
119  
(Pdb) host_ans
<MultiAsicSonicHost xxx>
(Pdb) dut_asic
<SonicAsic 0>
(Pdb) type(config_facts)
<class 'dict'>
(Pdb) config_facts.keys()
dict_keys(['ACL_TABLE', 'AUTO_TECHSUPPORT', 'AUTO_TECHSUPPORT_FEATURE', 'BGP_DEVICE_GLOBAL', 'BGP_NEIGHBOR', 'BGP_PEER_RANGE', 'BUFFER_PG', 'BUFFER_POOL', 'BUFFER_PROFILE', 'BUFFER_QUEUE', 'CABLE_LENGTH', 'CONSOLE_SWITCH', 'CRM', 'DEVICE_METADATA', 'DEVICE_NEIGHBOR', 'DEVICE_NEIGHBOR_METADATA', 'DHCP_SERVER', 'DSCP_TO_TC_MAP', 'FEATURE', 'FLEX_COUNTER_TABLE', 'GNMI', 'KDUMP', 'LOGGER', 'LOOPBACK_INTERFACE', 'MAP_PFC_PRIORITY_TO_QUEUE', 'MGMT_INTERFACE', 'MGMT_PORT', 'NTP_SERVER', 'PASSW_HARDENING', 'PFC_WD', 'PORT', 'PORTCHANNEL', 'PORTCHANNEL_INTERFACE', 'PORTCHANNEL_MEMBER', 'PORT_QOS_MAP', 'QUEUE', 'RESTAPI', 'SCHEDULER', 'SNMP', 'SNMP_COMMUNITY', 'SYSLOG_CONFIG', 'SYSLOG_CONFIG_FEATURE', 'SYSLOG_SERVER', 'SYSTEM_DEFAULTS', 'TACPLUS_SERVER', 'TC_TO_PRIORITY_GROUP_MAP', 'TC_TO_QUEUE_MAP', 'TELEMETRY', 'VERSIONS', 'VLAN', 'VLAN_INTERFACE', 'VLAN_MEMBER', 'WRED_PROFILE', 'port_name_to_alias_map', 'port_alias_to_name_map', 'port_index_map'])
(Pdb) 
```
Test result on T1:

```
=== Running tests individually ===
Running: python3 -m pytest snappi_tests/pfc/test_pfc_pause_lossless_with_snappi.py::test_pfc_pause_single_lossless_prio --inventory ../ansible/ixia,../ansible/veos --host-pattern STR-7050CX3-C19-U37 --testbed vms20-rdma-t0-7050cx3 --testbed_file /data/sonic-mgmt-int/ansible/testbed.yaml --log-cli-level warning --log-file-level debug --kube_master unset --showlocals --assert plain --show-capture no -rav --allow_recover --ignore=ptftests --ignore=acstests --ignore=saitests --ignore=scripts --ignore=k8s --ignore=sai_qualify --log-file logs/snappi_tests/pfc/test_pfc_pause_lossless_with_snappi.py::test_pfc_pause_single_lossless_prio.log --junitxml=logs/snappi_tests/pfc/test_pfc_pause_lossless_with_snappi.py::test_pfc_pause_single_lossless_prio.xml --topology tgen --skip_sanity
============================= test session starts ==============================
platform linux -- Python 3.8.10, pytest-7.4.0, pluggy-1.5.0
ansible: 2.13.13
rootdir: /data/sonic-mgmt-int/tests
configfile: pytest.ini
plugins: metadata-3.1.1, allure-pytest-2.8.22, forked-1.6.0, repeat-0.9.3, html-4.1.1, ansible-4.0.0, xdist-1.28.0
collected 2 items

snappi_tests/pfc/test_pfc_pause_lossless_with_snappi.py::test_pfc_pause_single_lossless_prio[xxx|3]
-------------------------------- live log call ---------------------------------
03:39:57 connection._warn                         L0246 WARNING| Verification of certificates is disabled
03:39:57 connection._warn                         L0246 WARNING| Unable to connect to http://10.64.246.150:443.
PASSED                                                                   [ 50%]
snappi_tests/pfc/test_pfc_pause_lossless_with_snappi.py::test_pfc_pause_single_lossless_prio[xxx|4] PASSED [100%]

=============================== warnings summary ===============================
../../../usr/local/lib/python3.8/dist-packages/paramiko/transport.py:236
  /usr/local/lib/python3.8/dist-packages/paramiko/transport.py:236: CryptographyDeprecationWarning: Blowfish has been deprecated
    "class": algorithms.Blowfish,

snappi_tests/pfc/test_pfc_pause_lossless_with_snappi.py::test_pfc_pause_single_lossless_prio[STR-7050CX3-C19-U37|3]
  /usr/local/lib/python3.8/dist-packages/snappi_ixnetwork/device/utils.py:2: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.10 it will stop working
    from collections import namedtuple, Mapping

snappi_tests/pfc/test_pfc_pause_lossless_with_snappi.py::test_pfc_pause_single_lossless_prio[STR-7050CX3-C19-U37|3]
snappi_tests/pfc/test_pfc_pause_lossless_with_snappi.py::test_pfc_pause_single_lossless_prio[STR-7050CX3-C19-U37|3]
  /usr/local/lib/python3.8/dist-packages/ixnetwork_restpy/testplatform/sessions/sessions.py:59: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    elif LooseVersion(build_number) < LooseVersion('8.52'):

snappi_tests/pfc/test_pfc_pause_lossless_with_snappi.py::test_pfc_pause_single_lossless_prio[STR-7050CX3-C19-U37|4]
  /usr/lib/python3/dist-packages/urllib3/connectionpool.py:1004: InsecureRequestWarning: Unverified HTTPS request is being made to host '10.64.246.150'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
- generated xml file: /data/sonic-mgmt-int/tests/logs/snappi_tests/pfc/test_pfc_pause_lossless_with_snappi.py::test_pfc_pause_single_lossless_prio.xml -
================== 2 passed, 5 warnings in 514.96s (0:08:34) ===================

```